workitem/workitem_revision_repository.go

```diff
@@ -0,0 +1,68 @@
+package workitem
+
+import (
+	"context"
+
+	"fmt"
+
+	"github.com/almighty/almighty-core/app"
+	"github.com/almighty/almighty-core/errors"
+	"github.com/almighty/almighty-core/log"
+	"github.com/jinzhu/gorm"
+	uuid "github.com/satori/go.uuid"
+)
+
+// WorkItemRevisionRepository encapsulates storage & retrieval of historical versions of work items
+type WorkItemRevisionRepository interface {
+	// TODO: use a 'Revision' structure to hold all parameters (except Context) ?
+	// Store stores a new revision for the given work item.
+	Store(ctx context.Context, modifierID uuid.UUID, operationType int, workitem WorkItem) error
+	// List retrieves all revisions for a given work item
+	List(ctx context.Context, workitem app.WorkItem) ([]WorkItemRevision, error)
```

Comment:
ok, that makes sense, indeed. No need to load the workitem from the DB, actually.